src/components/Profile.svelte

```diff
@@ -1,17 +1,104 @@
 <script>
+  import { db } from "../utils/api";
   import { handleLogout } from "../utils/auth";
-  import { useNavigate, useLocation } from "svelte-navigator";
   import { user } from "../utils/stores";
+  import { getAuth, updatePassword, onAuthStateChanged } from "firebase/auth";
+  import {
+    onSnapshot,
+    collection,
+    deleteDoc,
+    doc,
+    updateDoc,
+  } from "firebase/firestore";
+  import { update } from "firebase/database";
+  import UpdateListings from "./UpdateListings.svelte";
+
+  const auth = getAuth();
+
+  let fireUser;
+
+  onAuthStateChanged(auth, (firebaseUser) => {
+    if (firebaseUser) {
+      fireUser = firebaseUser;
+    } else {
+      console.log("no user!");
+    }
+  });
 
   let signedIn;
 
   user.subscribe((value) => {
     signedIn = value;
   });
 
+  let newPassword = "";
+
+  function changePassword() {
+    console.log("inside function");
+    updatePassword(fireUser, newPassword)
+      .then(() => {
+        newPassword = "";
+        console.log("Password Changed");
+      })
+      .catch((error) => {
+        console.log(error, "errror");
+      });
+  }
+
+  let myListings = [];
+
+  const getListings = onSnapshot(
+    collection(db, "listings"),
+    (querySnapshot) => {
+      let listingArray = [];
+      querySnapshot.forEach((listing) => {
+        let listingData = { ...listing.data(), id: listing.id };
+        listingArray = [listingData, ...listingArray];
+      });
+
+      myListings = listingArray.filter((listItem) => {
+        return listItem.user_id == signedIn.uid;
+      });
+    }
+  );
+
+  function deleteListing(listing) {
+    deleteDoc(doc(db, "listings", listing));
```

Comment:
Not essential, just suggestion of how app is currently structured - update and delete listings could be extracted to the API file and then imported into update listing to neaten it up rather than passing down as props into variables.